readthedocs/docsitalia/views/api.py

```diff
@@ -0,0 +1,43 @@
+# -*- coding: utf-8 -*-
+"""Docs italia api"""
+from __future__ import absolute_import
+
+from rest_framework import serializers, viewsets
+from rest_framework import mixins
+
+from readthedocs.projects.models import Project
+
+
+class ProjectsByTagSerializer(serializers.HyperlinkedModelSerializer):
+
+    """Projects by Tag DRF Serializer"""
+
+    class Meta:
+        model = Project
+        fields = ('name', 'description', 'url')
+
+
+class ProjectsByTagViewSet(
+    mixins.ListModelMixin,
+    viewsets.GenericViewSet
+):  # pylint: disable=too-many-ancestors
+
+    """
+    Projects by Tag DRF list model mixin
+
+    Takes a GET with tags separated by a comma:
+    tags=tag1,tag2
+    """
+
+    serializer_class = ProjectsByTagSerializer
+    allowed_methods = ('GET')
+
+    def get_queryset(self):
+        """filter projects by tags"""
+        tags = self.request.query_params.get('tags', None)
+        if tags:
+            tags = tags.split(',')
+            queryset = Project.objects.filter(tags__name__in=tags).distinct()
```

Comment:
slug and not name, no need for distinct, slugs should be unique

Reply:
ah the distinct is on the projects with multiple matched tags, sorry!

Reply:
I was following  taggit docs http://django-taggit.readthedocs.io/en/latest/api.html#filtering

Reply:
That's an example with plain orm though, we need something that won't be break things in urls :)